src/pages/programIndicators/form/AnalyticsPeriodBoundariesField.tsx

```diff
@@ -0,0 +1,351 @@
+import i18n from '@dhis2/d2-i18n'
+import {
+    Button,
+    ButtonStrip,
+    Field,
+    InputField,
+    Modal,
+    ModalActions,
+    ModalContent,
+    ModalTitle,
+    SingleSelectField,
+    SingleSelectOption,
+} from '@dhis2/ui'
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import { useField } from 'react-final-form'
+import { PeriodTypeSelect } from '../../../components/metadataFormControls/PeriodTypeSelect/PeriodTypeSelect'
+import { AnalyticsPeriodBoundary } from '../../../types/generated'
+import styles from '../../dataSetsWip/form/dataInputPeriods/DataInputPeriodsSelector.module.css'
+import classes from './AnalyticsPeriodBoundaries.module.css'
+import offsetPeriodType = AnalyticsPeriodBoundary.offsetPeriodType
+
+const boundaryTargets = [
+    {
+        label: i18n.t('Incident date'),
+        value: 'INCIDENT_DATE',
+    },
+    {
+        label: i18n.t('Event date'),
+        value: 'EVENT_DATE',
+    },
+    {
+        label: i18n.t('Enrollment date'),
+        value: 'ENROLLMENT_DATE',
+    },
+    {
+        label: i18n.t('Custom'),
+        value: 'CUSTOM',
+    },
+]
+
+const boundaryTypes = [
+    {
+        label: i18n.t('Before start of reporting period'),
+        value: AnalyticsPeriodBoundary.analyticsPeriodBoundaryType
+            .BEFORE_START_OF_REPORTING_PERIOD,
+    },
+    {
+        label: i18n.t('Before end of reporting period'),
+        value: AnalyticsPeriodBoundary.analyticsPeriodBoundaryType
+            .BEFORE_END_OF_REPORTING_PERIOD,
+    },
+    {
+        label: i18n.t('After start of reporting period'),
+        value: AnalyticsPeriodBoundary.analyticsPeriodBoundaryType
+            .AFTER_START_OF_REPORTING_PERIOD,
+    },
+    {
+        label: i18n.t('After end of reporting period'),
+        value: AnalyticsPeriodBoundary.analyticsPeriodBoundaryType
+            .AFTER_END_OF_REPORTING_PERIOD,
+    },
+]
+
+const APBItem = ({
+    apb,
+    onRemove,
+    onEdit,
+}: {
+    apb: Partial<AnalyticsPeriodBoundary>
```

Comment:
Especially this since its a named options property.
```suggestion
    periodBoundary: Partial<AnalyticsPeriodBoundary>
```